### ISSUE TITLE
Replace ``netifaces`` with ``ifaddr``

### ISSUE THREAD

**Issue by edavalosanaya** [ChimeraPy/Engine#214]
The ``ifaddr`` package provides the same functionality as ``netifaces``, but without the compilation or conda installation requirement.

**Comment by edavalosanaya** [comment]
```python
def get_ip_address() -> str:
    # Get all the network adapters
    adapters = ifaddr.get_adapters()
    
    # Iterate through the adapters to find a suitable IP
    for adapter in adapters:
        for ip in adapter.ips:
            # Check if it's an IPv4 address and not the local address
            if isinstance(ip.ip, str) and ip.ip != '127.0.0.1':
                return ip.ip

    # Return the local address if no suitable IP is found
    logger.warning("ChimeraPy-Engine: Couldn't find connected network, using 127.0.0.1")
    return "127.0.0.1"

```

**Comment by edavalosanaya** [comment]
Closed by PR #224
